spec/models/authenticator/httpd_spec.rb

```diff
@@ -707,17 +719,55 @@ def authenticate
         let(:headers) do
           super().merge('X-Remote-User-Groups' => 'wibble@fqdn,bubble@fqdn')
         end
+
+        it "handles a comma separated grouplist" do
+          expect(subject).to receive(:find_external_identity).with(username, user_attrs, ["wibble@fqdn", "bubble@fqdn"])
+          authenticate
+        end
+      end
+
+      context "using custom delimiter in settings" do
+        let(:config) { {:httpd_role => true, :group_delimiters => ","} }
+        let(:headers) do
+          super().merge('X-Remote-User-Groups' => 'wibble:wobble@fqdn,bubble@fqdn')
+        end
+
+        it "handles a group with a colon" do
+          expect(subject).to receive(:find_external_identity).with(username, user_attrs, ["wibble:wobble@fqdn", "bubble@fqdn"])
+          authenticate
+        end
+      end
+
+      context "using custom delimiter in settings" do
+        let(:config) { {:httpd_role => true, :group_delimiters => "|"} }
+        let(:headers) do
+          super().merge('X-Remote-User-Groups' => 'wibble:wobble@fqdn|bubble@fqdn')
+        end
+
+        it "handles a group with a colon" do
+          expect(subject).to receive(:find_external_identity).with(username, user_attrs, ["wibble:wobble@fqdn", "bubble@fqdn"])
+          authenticate
+        end
+      end
+
+      context "using header custom delimiter (prioritize config over header)" do
+        let(:config) { {:httpd_role => true, :group_delimiters => "|"} }
+        let(:headers) do
+          super().merge('X-Remote-User-Groups' => 'wibble:wobble@fqdn|bubble@fqdn', 'X-REMOTE-USER-GROUP-DELIMITER' => ",")
```

Comment:
Minor, but the header case should match the other one, so this should probably be

```suggestion
          super().merge('X-Remote-User-Groups' => 'wibble:wobble@fqdn|bubble@fqdn', 'X-Remote-User-Group-Delimiter' => ",")
```